src/Rules/DateTimeInstantiationRule.php

```diff
@@ -0,0 +1,70 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules;
+
+use DateTime;
+use PhpParser\Node;
+use PhpParser\Node\Expr\New_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Type\Constant\ConstantStringType;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\New_>
+ */
+class DateTimeInstantiationRule implements \PHPStan\Rules\Rule
+{
+
+	public function getNodeType(): string
+	{
+		return New_::class;
+	}
+
+	/**
+	 * @param New_ $node
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (
+			!($node->class instanceof \PhpParser\Node\Name)
+			|| \count($node->args) === 0
+			|| !\in_array(strtolower((string) $node->class), ['datetime', 'datetimeimmutable'], true)
+		) {
+			return [];
+		}
+
+		$arg = $scope->getType($node->args[0]->value);
+		if (!($arg instanceof ConstantStringType)) {
+			return [];
+		}
+
+		$errors = [];
+		$dateString = $arg->getValue();
+		try {
+			new DateTime($dateString);
+		} catch (\Throwable $e) {
+			// an exception is thrown for errors only but we want to catch warnings too
+		}
+		$lastErrors = DateTime::getLastErrors();
+		if ($lastErrors !== false) {
```

Comment:
You can get rid of these errors by adding `|false` to the method in `resources/functionMap.php` :) And after that I'll be able to merge it.

Reply:
Would this help https://github.com/JetBrains/phpstorm-stubs/pull/963?

Reply:
@b1rdex In some cases these stubs are used, but mostly it's a matter of these files: https://github.com/phpstan/phpstan-src/tree/master/resources

Reply:
I've updated functionMap